### PR TITLE
Merge dict #15

### PR DESCRIPTION
**Pull request by davidemoro**
Allow dict variables merge when multiple variable files is provided

### CODE REVIEW

File: test_variables.py [hunk 1]
```diff
@@ -109,3 +109,30 @@ def test(variables):
...
+        {'capabilities': {'browser': 'Firefox', 'browser_version': '53.0'}},
+        {'capabilities': {'debug': 'true'}}])
+    assert result.ret == 0
+
+
+def test_multiple_variables_merge_not_override_lists(testdir, file_format):
+    """ no lists extension, last wins """
+    testdir.makepyfile("""
+        def test(variables):
+            assert variables['list'] == [4, 5]
```
Comment (davidemoro) [inline test_variables.py]: @davehunt: do you prefer the extension of both lists [1,2,3,4]?

File: pytest_variables/plugin.py [hunk 2]
```diff
@@ -49,23 +50,49 @@ def pytest_addoption(parser):
...
+            variables = import_parser(path, *parser_table['json'])
         except ValueError as e:
             raise errors.ValueError('Unable to parse {0}: {1}'.format(
                 path, e))
 
+        if not isinstance(variables, dict):
+            raise errors.ValueError('Unable to parse {0}'.format(
+                path))
+
+        reduce(_merge, [config._variables, variables])
```
Comment (davidemoro) [inline pytest_variables/plugin.py]: we can avoid the loop ant let the reduce update config._variables with just one call ``reduce(_merge, [config._variables, variables1, ...])``. I decided to use a loop in order to keep things backwards compatible with existing tests

File: pytest_variables/plugin.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import sys
 
 import pytest
+from functools import reduce
```
Comment (davidemoro) [inline pytest_variables/plugin.py]: not necessary in python2

File: test_variables.py (hunk 1 above)
Comment (davehunt) [inline test_variables.py]: I agree with your implementation, and it's best to replace rather than extend.

**Comment by davidemoro** [comment]
Hi @davehunt, could you review the updated readme please?

**Comment by davehunt** [comment]
@davidemoro merged, and I've pushed a tag so there should be a v1.7.0 available shortly. 🤞

**Comment by davehunt** [comment]
I spoke too soon. Need to fix up Travis CI.

**Comment by davehunt** [comment]
Okay.. 🤞 https://travis-ci.org/pytest-dev/pytest-variables/builds/262194124

**Comment by davidemoro** [comment]
@davehunt I'm sorry, is there any problem with the PR?

**Comment by davehunt** [comment]
Boo.. "Skipping a deployment with the pypi provider because this branch is not permitted". I'll have to work it out. No problem with the PR at all.

**Comment by davehunt** [comment]
I've raised https://github.com/travis-ci/travis-ci/issues/8221 to understand the issue, but I'll push a deployment manually for now.

**Comment by davehunt** [comment]
Released! https://pypi.python.org/pypi/pytest-variables Thanks for the new feature @davidemoro!

**Comment by davidemoro** [comment]
Thanks!

**Pull request merged**